src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionposthearingapplication/ActionPostHearingApplicationSubmittedHandler.java

```diff
@@ -0,0 +1,111 @@
+import uk.gov.hmcts.reform.sscs.util.SscsUtil;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class ActionPostHearingApplicationSubmittedHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+    private final CcdService ccdService;
+    private final IdamService idamService;
+    @Value("${feature.postHearings.enabled}")
+    private final boolean isPostHearingsEnabled;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        requireNonNull(callback, "callback must not be null");
+        requireNonNull(callbackType, "callbacktype must not be null");
+
+        return callbackType.equals(CallbackType.SUBMITTED)
+            && callback.getEvent() == EventType.ACTION_POST_HEARING_APPLICATION
+            && isPostHearingsEnabled;
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback,
+                                                          String userAuthorisation) {
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(caseData);
+
+        Long caseId = Long.valueOf(caseData.getCcdCaseId());
+        if (!SscsUtil.isSAndLCase(caseData)) {
+            log.info("Action Post Hearing Application: Cannot process non Scheduling & Listing Case for Case ID {}",
+                caseId);
+            response.addError("Cannot process Action Post Hearing Application on non Scheduling & Listing Case");
+            return response;
+        }
+
+        ActionPostHearingApplication actionPostHearing = caseData.getActionPostHearingApplication();
+        ActionPostHearingTypes typeSelected = actionPostHearing.getTypeSelected();
+        log.info("Action Post Hearing Application: handing actionPostHearing {} for case {}", typeSelected,  caseId);
+
+        CcdCallbackMap action = getCcdCallbackMap(actionPostHearing, typeSelected);
+
+        if (isNull(action)) {
+            response.addError(String.format("Invalid Action Post Hearing Application Type Selected %s or action "
+                    + "selected as callback is null",
+                typeSelected));
+            return response;
+        }
+
+        if (nonNull(action.getCallbackEvent())) {
+            SscsCaseDetails updateCaseDetails = ccdService.updateCase(
+                caseData, caseId, action.getCallbackEvent().getCcdType(), action.getCallbackSummary(),
+                action.getCallbackDescription(), idamService.getIdamTokens());
+            return new PreSubmitCallbackResponse<>(updateCaseDetails.getData());
+        }
+
+        return response;
+    }
+
+    @Nullable
+    private static CcdCallbackMap getCcdCallbackMap(ActionPostHearingApplication actionPostHearing,
+                                                    ActionPostHearingTypes typeSelected) {
+        if (SET_ASIDE == typeSelected) {
```

Comment:
(m-ish) this would probably be neater as a switch or in the enum itself

Reply:
Changed to a switch :)